src/vmobjects/VMVector.cpp

```diff
@@ -0,0 +1,189 @@
+
+        storage->CopyIndexableFieldsTo(newStorage);
+        newStorage->SetIndexableField(last - 1, value);
+
+        SetField(2 /* storage */, newStorage);
+
+    } else {  // Just set the new value
+        storage->SetIndexableField(last - 1, value);
+    }
+
+    last += 1;
+    this->last = store_ptr(this->last, NEW_INT(last));
+}
+
+vm_oop_t VMVector::RemoveLast() {
+    const int64_t last = INT_VAL(load_ptr(this->last));
+    const int64_t first = INT_VAL(load_ptr(this->first));
+
+    // Throw an error correctly (error: method in som)
+    if (last == first) {
+        // VMSafe*Primitive::Invoke will push it right back to the same frame
+        VMFrame* frame = Interpreter::GetFrame();
+        vm_oop_t errorMsg =
+            Universe::NewString("Vector: error when removing Last item.");
+        vm_oop_t args[1] = {errorMsg};
+        this->Send("error:", args, 1);
+        return frame->Pop();
+    }
+    return Remove(NEW_INT(last - first));  // Last-First gives the (User) index
+                                           // of the last element in 1-indexing
+}
+
+vm_oop_t VMVector::RemoveFirst() {
+    // This method will just increment the first index
+    int64_t first = INT_VAL(load_ptr(this->first));
+
+    // Throw an error correctly (error: method in som)
+    if (first >= INT_VAL(load_ptr(this->last))) {
+        // VMSafe*Primitive::Invoke will push it right back to the same frame
+        VMFrame* frame = Interpreter::GetFrame();
+        vm_oop_t errorMsg =
+            Universe::NewString("Vector: error when removing First item.");
+        vm_oop_t args[1] = {errorMsg};
+        this->Send("error:", args, 1);
+        return frame->Pop();
+    }
+    vm_oop_t itemToRemove = GetIndexableField(
+        1);      // This is 1 because GetIndexableField handles 1 to 0 indexing
+    first += 1;  // Increment the first index
+    this->first = store_ptr(this->first, NEW_INT(first));
+    return itemToRemove;
+}
+
+vm_oop_t VMVector::RemoveObj(vm_oop_t other) {
+    const int64_t first = INT_VAL(load_ptr(this->first));
+    const int64_t last = INT_VAL(load_ptr(this->last));
+    VMArray* storage = load_ptr(this->storage);
+
+    for (int64_t i = first - 1; i < last - 1; ++i) {
+        vm_oop_t current = storage->GetIndexableField(i);
+
+        // Check where integers are tagged or references can be checked
+        if (current == other) {
+            Remove(NEW_INT(i - first + 2));  // Convert to 1-indexing
+            return load_ptr(trueObject);
+        }
+    }
+    return load_ptr(falseObject);
+}
+
+vm_oop_t VMVector::Remove(vm_oop_t inx) {
+    const int64_t first = INT_VAL(load_ptr(this->first));
+    int64_t last = INT_VAL(load_ptr(this->last));
+    VMArray* storage = load_ptr(this->storage);
+    int64_t index = INT_VAL(inx);
+
+    if ((last - first) != 0 && index == 0) {
+        index = index;
+    }
+
+    if (index < 1 || index > last - first) {
+        IndexOutOfBounds();
+    }
```

Comment:
This code here is a bit odd. I think `Remove(inx)` is not used anywhere else, and allocating an `int` here seems unnecessary for a helper.
But the code is also too general, I think.
If this is truly only a helper for `RemoveObj()` (if I didn't miss anything), then it doesn't need to handle the failure case, i.e., `IndexOutOfBounds`